extensions/csharp-v2/operation/method.ts

```diff
@@ -157,7 +157,7 @@ export class OperationMethod extends Method {
         yield `// verify that Identity format is an exact match for uri`;
         yield EOL;
 
-        const match = Local("_match", `${System.Text.RegularExpressions.Regex.new(rx).value}.Match(${identity.value})`);
+        const match = Local("_match", `${System.Text.RegularExpressions.Regex.new(rx).value}.Match(${_identity.value})`);
```

Comment:
Not fully understanding the viaIdentity path, I'm curious if this changes the behavior of the generated code. Can you share the whole before/after generated method?

Reply:
I've changed the name of the path parameter to `EXAMPLE` to make it more readable.

 Note that if the user-provided parameter were to be `identity`, the collision would occur on the `var example = _match.Groups["EXAMPLE"].Value;` line.

**WITHOUT CHANGE**
```c#
public async global::System.Threading.Tasks.Task TeamsCallParkPolicyGetByIdViaIdentity(global::System.String identity, string Select, global::System.Func<global::System.Net.Http.HttpResponseMessage, global::System.Threading.Tasks.Task<Sample.API.Models.ISkypeForBusinessTeamsCallParkPolicy>, global::System.Threading.Tasks.Task> onOk, global::System.Func<global::System.Net.Http.HttpResponseMessage, global::System.Threading.Tasks.Task<Sample.API.Models.IError>, global::System.Threading.Tasks.Task> onDefault, Sample.API.Runtime.IEventListener eventListener, Sample.API.Runtime.ISendAsync sender)
{
    // Constant Parameters
    using( NoSynchronizationContext )
    {
        // verify that Identity format is an exact match for uri
        var _match = new global::System.Text.RegularExpressions.Regex("^/TeamsCallParkPolicy('(?<EXAMPLE>[^/]+)')$").Match(identity);
        if (!_match.Success)
        {
            throw new global::System.Exception("Invalid identity for URI '/TeamsCallParkPolicy('{EXAMPLE}')'");
        }
        // replace URI parameters with values from identity
        var example = _match.Groups["EXAMPLE"].Value;
        // construct URL
        var _url = new global::System.Uri((
                "http://default/odataTeamsCallParkPolicy('"
                + example
                + "')"
                + "?"
                + (string.IsNullOrEmpty(Select) ? global::System.String.Empty : "$select=" + System.Uri.EscapeDataString(Select))
                ).TrimEnd('?','&'));
        await eventListener.Signal(Sample.API.Runtime.Events.URLCreated, _url); if( eventListener.Token.IsCancellationRequested ) { return; }
        // generate request object
        var request =  new global::System.Net.Http.HttpRequestMessage(Sample.API.Runtime.Method.Get, _url);
        await eventListener.Signal(Sample.API.Runtime.Events.RequestCreated, _url); if( eventListener.Token.IsCancellationRequested ) { return; }
        await eventListener.Signal(Sample.API.Runtime.Events.HeaderParametersAdded, _url); if( eventListener.Token.IsCancellationRequested ) { return; }
        // make the call
        await this.TeamsCallParkPolicyGetById_Call(request,onOk,onDefault,eventListener,sender);
    }
}
```

**WITH CHANGE**
```c#
public async global::System.Threading.Tasks.Task TeamsCallParkPolicyGetByIdViaIdentity(global::System.String _identity, string Select, global::System.Func<global::System.Net.Http.HttpResponseMessage, global::System.Threading.Tasks.Task<Sample.API.Models.ISkypeForBusinessTeamsCallParkPolicy>, global::System.Threading.Tasks.Task> onOk, global::System.Func<global::System.Net.Http.HttpResponseMessage, global::System.Threading.Tasks.Task<Sample.API.Models.IError>, global::System.Threading.Tasks.Task> onDefault, Sample.API.Runtime.IEventListener eventListener, Sample.API.Runtime.ISendAsync sender)
{
    // Constant Parameters
    using( NoSynchronizationContext )
    {
        // verify that Identity format is an exact match for uri
        var _match = new global::System.Text.RegularExpressions.Regex("^/TeamsCallParkPolicy('(?<EXAMPLE>[^/]+)')$").Match(_identity);
        if (!_match.Success)
        {
            throw new global::System.Exception("Invalid identity for URI '/TeamsCallParkPolicy('{EXAMPLE}')'");
        }
        // replace URI parameters with values from identity
        var example = _match.Groups["EXAMPLE"].Value;
        // construct URL
        var _url = new global::System.Uri((
                "http://default/odataTeamsCallParkPolicy('"
                + example
                + "')"
                + "?"
                + (string.IsNullOrEmpty(Select) ? global::System.String.Empty : "$select=" + System.Uri.EscapeDataString(Select))
                ).TrimEnd('?','&'));
        await eventListener.Signal(Sample.API.Runtime.Events.URLCreated, _url); if( eventListener.Token.IsCancellationRequested ) { return; }
        // generate request object
        var request =  new global::System.Net.Http.HttpRequestMessage(Sample.API.Runtime.Method.Get, _url);
        await eventListener.Signal(Sample.API.Runtime.Events.RequestCreated, _url); if( eventListener.Token.IsCancellationRequested ) { return; }
        await eventListener.Signal(Sample.API.Runtime.Events.HeaderParametersAdded, _url); if( eventListener.Token.IsCancellationRequested ) { return; }
        // make the call
        await this.TeamsCallParkPolicyGetById_Call(request,onOk,onDefault,eventListener,sender);
    }
}
```

Running a diff on these two methods, the only change is `identity` to `_identity`.

Reply:
Chatted with @joshspicer offline - there's a number of ways the existing parameters could still collide (rest parameter called _identity, or body, etc.).

Makes more sense to rename the other end of the collision, because the method parameters are all known in advance, but the REST parameters aren't. Calling them something like "rest_identity" should be safe, since none of the method parameters will have such a prefix.

Reply:
Generated method now looks like this (for given path parameter `EXAMPLE`).
```c#
...
 // replace URI parameters with values from identity
 var rest_example = _match.Groups["EXAMPLE"].Value;
 // construct URL
 var _url = new global::System.Uri((
         "http://default/odataTeamsCallParkPolicy('"
         + rest_example
         + "')"
         + "?"
         + (string.IsNullOrEmpty(Select) ? global::System.String.Empty : "$select=" + System.Uri.EscapeDataString(Select))
         ).TrimEnd
...
```